Client/GeneralizedImageFetcher.swift

```diff
@@ -0,0 +1,49 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import UIKit
+import Shared
+import Logger
+
+struct GeneralizedImageFetcher {
+    /// This method is a generalized image fetcher that'll also cache. The completion is dispatched to `main`.
+    func getImageFor(url: URL, completion: @escaping (UIImage?) -> Void) {
```

Comment:
Could we use [swift doc format](https://sarunw.com/posts/swift-documentation/) for comments on that function? You can generate them with `option + cmd + /` when your cursor is on top of the function.